client/src/lib.rs

```diff
@@ -0,0 +1,486 @@
+//! Turnkey Client to interact with the Turnkey API
+//! See <https://docs.turnkey.com>
+use std::time::{SystemTime, UNIX_EPOCH};
+use thiserror::Error;
+
+use generated::google::rpc::Status;
+use generated::Activity;
+use generated::ActivityResponse;
+use generated::ActivityStatus;
+
+use tkhq_api_key_stamper::TurnkeyApiKey;
+
+#[cfg_attr(doc, doc(hidden))]
+pub mod generated;
+
+pub mod retry;
+pub use retry::RetryConfig;
+
+#[derive(Debug, Error)]
+pub enum TurnkeyClientError {
+    #[error("HTTP request failed: {0}")]
+    Http(#[from] reqwest::Error),
+
+    #[error("Failed to decode response {0} ({1})")]
+    Decode(String, serde_json::Error),
+
+    #[error("Serde JSON failure: {0}")]
+    SerdeJsonFailure(#[from] serde_json::Error),
+
+    #[error("Missing activity from response")]
+    MissingActivity,
+
+    #[error("Missing result from activity")]
+    MissingResult,
+
+    #[error("Missing inner result from activity result")]
+    MissingInnerResult,
+
+    #[error("Activity status unexpected: {0}")]
+    UnexpectedActivityStatus(String),
+
+    #[error("Unexpected inner activity result: {0}")]
+    UnexpectedInnerActivityResult(String),
+
+    #[error("Activity failed processing: {0:?}")]
+    ActivityFailed(Option<Status>),
+
+    #[error("This activity ({0}) requires consensus and needs extra approvals")]
+    ActivityRequiresApproval(String),
+
+    #[error("Maximum number of attempts reached (after {0} retries)")]
+    ExceededRetries(usize),
+}
+
+pub struct TurnkeyClient {
+    http: reqwest::Client,
+    base_url: String,
+    api_key: TurnkeyApiKey,
+    retry_config: RetryConfig,
```

Comment:
nit: this could be an Option, and use the default if not provided

Reply:
Great idea!